cobbler/actions/buildiso/__init__.py

```diff
@@ -0,0 +1,242 @@
+"""
+Builds bootable CD images that have PXE-equivalent behavior for all Cobbler distros/profiles/systems currently in
+memory.
+"""
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+# SPDX-FileCopyrightText: Copyright 2006-2009, Red Hat, Inc and Others
+# SPDX-FileCopyrightText: Michael DeHaan <michael.dehaan AT gmail>
+
+import logging
+import os
+import pathlib
+import shutil
+from typing import Dict, List, Optional, Union
+
+from cobbler import utils
+
+
+def add_remaining_kopts(kopts: dict) -> str:
+    """Add remaining kernel_options to append_line
+
+    :param kopts: The kernel options which are not present in append_line.
+    :return: A single line with all kernel options from the dictionary in the string. Starts with a space.
+    """
+    append_line = [""]  # empty str to ensure the returned str starts with a space
+    for option, args in kopts.items():
+        if args is None:
+            append_line.append(f"{option}")
+            continue
+
+        if not isinstance(args, list):
+            args = [args]
+
+        for arg in args:
+            arg_str = format(arg)
+            if ' ' in arg_str:
+                arg_str = f'"{arg_str}"'
```

Comment:
Are there any chances that one of the args can contain `"` inside and we get broken format string at the end?
I mean something like `onearg theoher="test" it"`

Reply:
This is indeed possible. So we need to safeguard it. Since I don't have something off my mind in 2 min I will do so after lunch!

Reply:
What's the consequence of having a `"` inside the returned `str`? Afaik it's used to not split the option=value pairs by "protecting" spaces inside the value. Misplaced `"` probably cause an unintended interpretation, is there anything else that can be caused here?

Right now this is returned by this function (lines 19-40):
```python
In [24]: kopts={"good_opt": "val", "bad_opt": 'val"', "theother": 'test" it'}

In [25]: add_remaining_kopts(kopts)
Out[25]: ' good_opt=val bad_opt=val" theother="test" it"'
```

Reply:
Okay. @meaksh @vzhestkov @agraul and myself agreed in a call that this function is behaving as desired and we document its limitations for the Cobbler users.